### PR TITLE
Add post about capsule updates

### PR DESCRIPTION
**Pull request by SergiiDmytruk**
- [x] Urls to MSI releases need to be updated.
- [x] One link points to documentation from https://github.com/Dasharo/docs/pull/915 which wasn't merged yet, but shouldn't change.

### CODE REVIEW

**Review by krystian-hebel — CHANGES_REQUESTED** [review]
Also, please change the target of this PR to `develop`.

File: blog/content/post/2024-10-11-uefi-capsule-updates.md [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+
+## Firmware updates
+
+Firmware is a crucial software component without which a device can't function
+properly, in fact, most likely can't function at all.  Like anything, firmware
+is subject to defects and, due to its integral role for a device, a firmware
+defect can have a significant impact on its operation.  A bad firmware can be
+subject to an attack or sometimes cause a physical damage to a device.  Even in
+the absence of any issues, an update might still be desirable to provide more
+configuration options or support wider range or hardware.
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2024-10-11-uefi-capsule-updates.md]: https://github.com/3mdeb/news-and-ideas/compare/03fea547337a8622769f09ba08647100f651ab70..fc56b56081baa6eafafea7e60a43adde55d7283b#diff-567b97bbd9f272b17256eb59986950bcfa0120ec4c9b5ab0b31c72c3c70242eeR52

File: blog/content/post/2024-10-11-uefi-capsule-updates.md [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+capsules are an example of moving in the direction of greater convenience to
+end users.
+
+## Firmware update methods
+
+Let's have a quick overview of update methods to have the necessary context for
+discussing capsule updates.
+
+First of all, an update can be performed either internally or externally.  An
+external update often requires a special device capable of updating a [flash
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2024-10-11-uefi-capsule-updates.md]: https://github.com/3mdeb/news-and-ideas/compare/03fea547337a8622769f09ba08647100f651ab70..fc56b56081baa6eafafea7e60a43adde55d7283b#diff-567b97bbd9f272b17256eb59986950bcfa0120ec4c9b5ab0b31c72c3c70242eeR67-R87

File: blog/content/post/2024-10-11-uefi-capsule-updates.md [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+chip] which contains the firmware.  Motherboards supporting updates while the
+machine is turned off essentially have such a device built into them.
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2024-10-11-uefi-capsule-updates.md]: https://github.com/3mdeb/news-and-ideas/compare/03fea547337a8622769f09ba08647100f651ab70..fc56b56081baa6eafafea7e60a43adde55d7283b#diff-567b97bbd9f272b17256eb59986950bcfa0120ec4c9b5ab0b31c72c3c70242eeR67-R87

File: blog/content/post/2024-10-11-uefi-capsule-updates.md [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+[tools-doc]: https://docs.dasharo.com/kb/edk2-capsule-updates/
+
+### Hardware testing
+
+Since MSI boards were the first to get the feature, that's where it was
+validated on hardware.  Manual tests are done as described by the aforementioned
+[update guide][update-guide], but that's not the real tests.  Real tests are
+run automatically via [Robot Framework][robot] and follow a similar sequence but
+also take into account various corner cases, verify whether data was actually
+migrated or that UX capsule changed how update process looks.  The [OSFV] tests
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2024-10-11-uefi-capsule-updates.md]: https://github.com/3mdeb/news-and-ideas/compare/03fea547337a8622769f09ba08647100f651ab70..fc56b56081baa6eafafea7e60a43adde55d7283b#diff-567b97bbd9f272b17256eb59986950bcfa0120ec4c9b5ab0b31c72c3c70242eeR280-R281

File: blog/content/post/2024-10-11-uefi-capsule-updates.md [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+A set of settings available in a given firmware is likely to remain the same or
+grow in a more recent version.  It's nice to not have to recreate the
+same configuration after every update.  The same goes about custom boot logo and
+other user-configurable parts of a firmware.
+
+In practice this means copying data from `SMMSTORE`, `COREBOOT` and
+`BOOTSPLASH` regions into updated firmware so it's already suitably configured
+on the first boot.
+
+### UX capsules
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2024-10-11-uefi-capsule-updates.md]: https://github.com/3mdeb/news-and-ideas/compare/03fea547337a8622769f09ba08647100f651ab70..fc56b56081baa6eafafea7e60a43adde55d7283b#diff-567b97bbd9f272b17256eb59986950bcfa0120ec4c9b5ab0b31c72c3c70242eeR319-R339

File: blog/content/post/2024-10-11-uefi-capsule-updates.md [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+
+## Firmware update methods
+
+Let's have a quick overview of update methods to have the necessary context for
+discussing capsule updates.
+
+First of all, an update can be performed either internally or externally.  An
+external update often requires a special device (see [this kit][ec-kit] for an
+idea) capable of updating a [flash chip] which contains the firmware.  The
+device gets connected to a host machine (so you need a second PC/lapatop) from
```
Comment (krystian-hebel) [inline blog/content/post/2024-10-11-uefi-capsule-updates.md:72]:
```suggestion
device gets connected to a host machine (so you need a second PC/laptop) from
```

Comment (SergiiDmytruk) [reply blog/content/post/2024-10-11-uefi-capsule-updates.md]: https://github.com/3mdeb/news-and-ideas/compare/fc56b56081baa6eafafea7e60a43adde55d7283b..93c8836df2ed0f57e65ae72375ebc06e04d2b5b2

File: blog/content/post/2024-10-11-uefi-capsule-updates.md [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+for these boards.  Only future updates can leverage this update method unless
+you use it to "reinstall" newest releases as an experiment.
+
+Worth mentioning right away that this work was funded by [NLnet
+Foundation][nlnet] in the interest of making updates of open source firmware a
+more pleasant experience.  More details about the project can be found
+[here][nlnet-project].
+
+[z690]: https://docs.dasharo.com/variants/msi_z690/releases/#v114-2024-
+[z790]: https://docs.dasharo.com/variants/msi_z790/releases/#v092-2024-
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply blog/content/post/2024-10-11-uefi-capsule-updates.md]: https://github.com/3mdeb/news-and-ideas/compare/93c8836df2ed0f57e65ae72375ebc06e04d2b5b2..b5462cd775a3af6047e7ece34c7524c968f49cf7#diff-dfe25d351527b03db1f64dad949ef9983013786d096297d550f5433b8a83657dR41-R42

Also updated post's date.
